services/src/contexts/mod.rs

```diff
@@ -235,9 +236,10 @@ where
         let cfg = get_config_element::<crate::util::config::MachineLearning>()
             .map_err(|_| geoengine_operators::error::Error::InvalidMachineLearningConfig)?;
 
-        let model_base_path = cfg.model_defs_path;
+        let model_base_path = cfg.model_defs_path.canonicalize()?;
 
-        let model_path = canonicalize_subpath(&model_base_path, &model_sub_path)?;
+        let model_path =
+            canonicalize_subpath(&model_base_path, &model_sub_path)?.join("model.json");
```

Comment:
the `model_sub_path` should now be a `model_id`. It should also be of type `ModelId` which you can create using the `identifier!` macro

Reply:
1168be2810c99a6bbee7b5b4d26596dee1373a5a